### PR TITLE
Feature/metafields

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
**Are you a current Fivetran customer?** 

Fivetran created PR

**What change(s) does this PR introduce?** 

This PR adds two additional fields to the `stg_shopify__metafield` model. These fields are crucial for downstream metafield mapping purposes.

**Did you update the CHANGELOG?** 

- [X] Yes
The addition of the metafield model has already been included in the CHANGELOG. Therefore, there was no need to mark this entry.

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [X] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [X] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [ ] Yes, Issue/Feature [link bug/feature number here]
- [X] No 

**How did you test the PR changes?** 

 

- [X] Buildkite  
- [ ] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [X] BigQuery
- [X] Redshift
- [X] Snowflake
- [X] Postgres
- [X] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
🤘 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: models/stg_shopify__metafield.sql [hunk 1]
```diff
@@ -37,6 +37,8 @@ final as (
         {{ dbt_date.convert_timezone(column='cast(created_at as ' ~ dbt.type_timestamp() ~ ')', target_tz=var('shopify_timezone', "UTC"), source_tz="UTC") }} as created_at,
         {{ dbt_date.convert_timezone(column='cast(updated_at as ' ~ dbt.type_timestamp() ~ ')', target_tz=var('shopify_timezone', "UTC"), source_tz="UTC") }} as updated_at,
         {{ dbt_date.convert_timezone(column='cast(_fivetran_synced as ' ~ dbt.type_timestamp() ~ ')', target_tz=var('shopify_timezone', "UTC"), source_tz="UTC") }} as _fivetran_synced,
+        {{ dbt.concat(["namespace","'_'","key"]) }} as metafield_reference,
+        row_number() over (partition by id order by updated_at desc) = 1 as is_most_recent_record,
```
Comment (fivetran-jamie) [inline models/stg_shopify__metafield.sql:41]: i don't see any records where `is_most_recent_record` is false -- is this the correct partition or do we not need this?

**Review by fivetran-jamie — APPROVED** [review]
looks good -- just a question around `is_most_recent_record`  since it doesn't appear to ever be false (and we don't use it downstream?)

File: models/stg_shopify__metafield.sql (hunk 1 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_shopify__metafield.sql:41]: I noticed that there _may_ be cases where metafields can be updated so I chose to include this column and filter in the downstream macro. However, there were no updates to the data we have but I still felt it relevant to include.

Comment (fivetran-joemarkiewicz) [reply models/stg_shopify__metafield.sql:41]: Good catch on it not being used downstream. That was the intention. I will add it into the macro.

**Pull request merged**